cdn/cdn_test.go

```diff
@@ -0,0 +1,18 @@
+package cdn
+
+import (
+	"context"
+	"fmt"
+	"log"
+)
+
+func ExampleCheckNameAvailability() {
+	ctx := context.Background()
+
+	available, err := CheckNameAvailability(ctx, "gocdnname", "Microsoft.Cdn/Profiles/Endpoints")

```

Comment:
>gocdnname [](start = 47, length = 9)

Since this will be used also for testing, I would choose a name we are sure will be available.
I would add some randomness with `helpers.GetRandomLetterSequence()`

Reply:
Fixed in f8d0c81